editor/modes/visual-editor/block.js

```diff
@@ -251,12 +314,54 @@ class VisualEditorBlock extends Component {
 
 	onKeyDown( event ) {
 		const { keyCode, target } = event;
+		const { previousBlock, nextBlock, onFocus } = this.props;
+		const up = keyCode === UP;
+		const down = keyCode === DOWN;
+		const left = keyCode === LEFT;
+		const right = keyCode === RIGHT;
+
+		if ( up || down || left || right ) {
```

Comment:
Pyramid of `if` could be improved if we delegated this to some helper methods that had early returns (as previously with `handleArrowKey`). See below comment too where we have early return that will inadvertently skip other `onKeyDown` logic.